include/picongpu/plugins/openPMD/WriteSpecies.hpp

```diff
@@ -496,7 +497,7 @@ namespace picongpu
                         extent_x.resetDataset(ds);
 
                         // particleOffset[d] is allowed to be negative for the first GPU
-                        using OffsetType = decltype(particleOffset[d]);
+                        using OffsetType = std::remove_reference_t<decltype(particleOffset[d])>;
```

Comment:
I think `decltype` would also return a reference here e.g. for `std::vector`. Ofc, that one offers `::value_type` to directly get the element type, and this `Space` type is not defined enough to rely on it.

Reply:
Ah, I thought that `decltype` would decay the type the same way that `auto` does, but apparently not.